lib/paper_trail/record_trail.rb

```diff
@@ -414,13 +410,29 @@ def add_transaction_id_to(data)
       data[:transaction_id] = PaperTrail.transaction_id
     end
 
+    # Given a HABTM association, returns an array of ids.
+    # @api private
+    def habtm_assoc_ids(habtm_assoc)
+      current = @record.send(habtm_assoc.name).to_a.map(&:id) # TODO: `pluck` would use less memory
```

Comment:
> I believe in ActiveRecord you're able to call something like @record.send("#{habtm_assoc.name}_ids") to get the current list of ids ..

According to the [associations guide](http://guides.rubyonrails.org/association_basics.html#has-and-belongs-to-many-association-reference) we'd have to `singularize` the association name.

```ruby
@record.send("#{habtm_assoc.name.singularize}_ids")
```

Simply using `singularize` introduces the potential for a `NoMethodError` with irregular nouns unless properly configured in the `Inflector`. I looked for a method like `name_singular`, that would use the model config instead of relying on the inflector config, but I couldn't find one.